src/Component/CodeEditor/CodeEditor.tsx

```diff
@@ -126,23 +127,19 @@ export class CodeEditor extends React.Component<CodeEditorProps, CodeEditorState
   }
 
   getModeByParser = (): string => {
-    const {
-      activeParser
-    } = this.state;
-    if (activeParser && activeParser.title === 'SLD Style Parser') {
+    const activeParser: any = this.state.activeParser;
+    if (activeParser && (activeParser.title || activeParser.prototype.title) === 'SLD Style Parser') {
       return 'application/xml';
     }
     return 'application/json';
   }
 
   valueFromStyleInput = (style: GsStyle) => {
-    const {
-      activeParser
-    } = this.state;
+    const activeParser: any = this.state.activeParser;
     return new Promise((resolve, reject) => {
       if (activeParser) {
-        const StyleParser = activeParser;
-        const parserInstance = new StyleParser();
+        const StyleParserClass = activeParser;
+        const parserInstance = activeParser instanceof Function ? new StyleParserClass() : activeParser;
```

Comment:
This seems to be kind of a loose check. But not blocking actually.

Reply:
It is, but unfortunately the `instanceof StyleParser` cannot be used since it is an interface type.